terraform/utils.go

```diff
@@ -0,0 +1,61 @@
+package terraform
+
+import (
+	"sort"
+	"strings"
+)
+
+func isValidVariableFile(fileName string) bool {
+	if strings.HasSuffix(fileName, DEFAULT_TFVARS) {
+		return true
+	}
+	for _, fileExt := range VALID_VARIABLE_FILES {
+		if strings.HasSuffix(fileName, fileExt) {
+			return true
+		}
+	}
+	return false
+}
+
+func isValidTerraformFile(fileName string) bool {
+	for _, fileExt := range VALID_TERRAFORM_FILES {
+		if strings.HasSuffix(fileName, fileExt) {
+			return true
+		}
+	}
+	return false
+}
+
+func orderFilesByPriority(fileNames []string) []string {
```

Comment:
Nothing wrong with the current function but I do want to share another idea about how to implement it and discuss it.
It looks like we are going to end up with a bunch of loops and each loop will contain its own unique logic for the specific file type that we are looking for, what if instead of that we will do something like the following:
```
1. Separately create a new enum that will contain the file type and its priority (e.g. .tf = 1)
2. Sort the files lexical order
3. Iterate through the files and attach to each one of them its priority based on the file name
4. Sort the files by the priority while keeping the lexical order
5. Return the sorted list
```
This way when we would like to add support for a new file type we will only need to add its priority to the enum.
WDYT?

Reply:
I'll give it a go and see how readable it is, thanks! 👍